### ISSUE TITLE
GDPR Setting

### ISSUE THREAD

**Issue by sbulen** [SimpleMachines/SMF#6228]
#### Description
2.0 has a GDPR-enabled setting under Core Features.  When enabled, certain behaviors are enforced:
- Policy acceptance is required
- Reg acceptance is required
- Admin cannot specify default settings for users on announcements (users must opt-in)
- Newsletter – Admins are not allowed to override the user settings when sending newsletters
- Admins must allow users to disable announcements, this cannot be overridden

There is no functional equivalent in 2.1.  Should there be???

I think we need an equivalent in 2.1.  

Open to input...  Opening this issue to capture discussion & resolution.

**Comment by Oldiesmann** [comment]
I definitely think the behavior should be the same in 2.1 (and possibly apply to alerts/notifications as well? Not entirely sure how the GDPR works).

**Comment by sbulen** [comment]
There is some internal discussion on this, not sure it's settled.  

I think folks are concerned we shouldn't be locking folks in as rigidly as 2.0 does.  But the 2.0 function is nice for letting folks know they're covered.  Seeking a middle ground - like maybe VERY clearly labeling the settings.  

Some of the 2.1 behavior is already GDPR compliant where it is an option in 2.0.  (e.g., no-logon unsubscribe is 2.1 behavior)

I think all that's left are the various places admins can override user notification preferences.  Just 2-3 settings.

**Comment by LexArma** [comment]
I think, if we want to make it real easy and clear as day - then a master setting like that makes sense. 
On the other hand, if we only want to make the software compliant, and leave the actual interpreting of the (many times) national regulation up to the user, then all we need is a set of settings that can be combined so that GDPR demands are met.
Which is better, I'm actually a bit torn on this - but since each EU country has some leeway in interpreting the EU regulations, maybe it is better for us to just have the individual settings that may or may not reach GDPR requirements depending on the admin.

**Comment by MissAllSunday** [comment]
I'm on the idea that we should only provide the needed features to allow users to comply with regulations but we shouldn't enforce or otherwise decide for the user.

**Comment by LexArma** [comment]
Yes, that I agree with - But the few settings listed in the OP, are bare minimum GDPR features, and the question as I understood it wasn't really if should we enforce them, but should we provide a single setting to enable them all.

**Comment by MissAllSunday** [comment]
Moved to Final so we can have some more time to think this better.

**Comment by Sesquipedalian** [comment]
I don't think a one-click toggle is the best approach for SMF 2.1. 

> - Policy acceptance is required
> - Reg acceptance is required
> - Admin cannot specify default settings for users on announcements (users must opt-in)
> - Newsletter – Admins are not allowed to override the user settings when sending newsletters
> - ~Admins must allow users to disable announcements, this cannot be overridden~

The crossed-out item is always true in SMF 2.1, so that leaves only the rest to deal with.

The first two settings can't easily be addressed by a one-click toggle, because they require a registration agreement and privacy policy to exist before they can be enabled, but we don't provide a default privacy policy. (This is why we had to code some extra safety checks in 2.0 that prevented the admin from activating the one-click GPDR-compliance toggle until the privacy policy existed.)

We *could* implement a GDPR-compliance toggle in 2.1 to hide the "Override notification settings" option when sending newsletters and the "Receive forum newsletters, announcements and important notifications by email" in the default notification preferences. But that would be creating a checkbox in one part of the ACP whose only purpose was to hide a couple of other checkboxes somewhere else in the ACP, which would be pretty silly. 

Regarding the "Override notification settings" option when sending newsletters, I think we should either:

1. simply remove the "Override notification settings" option for sending newsletters, or
2. add warnings about breaking GDPR compliance to the "Override notification settings" option.

I'm in favour of removing the option entirely. From a development perspective, it is the simpler, cleaner solution. From a forum administration perspective, respecting users' notification preferences is *always* the best practice. Indeed, sending newsletters to members who have not agreed to "Receive forum newsletters, announcements and important notifications by email" is a quick way to piss them off and have them start flagging your emails as spam (I've done exactly that more than once myself). If some admin really wants to bring back the "Override notification settings" option when sending newsletters, I'm sure some enterprising mod author will provide it.

Regarding the  "Receive forum newsletters, announcements and important notifications by email" option in the default notification preferences, I think we should just add a warning in the setting description text that enabling this by default violates the GDPR and many other countries' digital privacy laws.

**Comment by sbulen** [comment]
Yes...  The internal discussion was leaning towards simply changing the help text & leaving all logic as is.  E.g.: "Recommended in some regions for compliance..."

**Comment by LexArma** [comment]
Actually, after some thought, I would suggest the warning is probably better. GDPR allows exceptions to the rule, for example notifying all users of a data breach.

**Comment by Sesquipedalian** [comment]
Fair enough, then. I'll revise #6807 in light of these preferences.

**Comment by Sesquipedalian** [comment]
Currently, the help text is where we advise the admin about the GDPR requirements for the "Require new members to accept the registration agreement" and "Require new members to accept the privacy policy" settings. Should we make that advice more prominent by moving it into a description line below each setting?

**Comment by Sesquipedalian** [comment]
There is something else that wasn't on @sbulen's list above.

In SMF 2.0, enabling GDPR compliance mode removes the ability to send newsletters to arbitrary email addresses. We did that because there is no way to issue unsubscribe tokens for non-member email addresses, which made it hard to follow the GDPR's requirement to include an unsubscribe option in the email message. But that isn't a viable solution for SMF 2.1. 

Instead, I've added a new set of unsubscribe instructions that will be appended to newsletters sent to non-member email addresses: 'To unsubscribe from forum newsletters, announcements and important notifications, contact us at %1$s with your request.' (The webmaster email address will be substituted into the string.) This allows us to keep the functionality while remaining compliant with the GDPR.
